extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala

```diff
@@ -17,12 +17,27 @@
 
 package org.apache.kyuubi.plugin.spark.authz.ranger
 
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+
+import org.apache.ranger.plugin.policyengine.RangerAccessRequest
 import org.apache.ranger.plugin.service.RangerBasePlugin
 
+import org.apache.kyuubi.plugin.spark.authz.AccessControlException
 import org.apache.kyuubi.plugin.spark.authz.util.AuthZUtils._
 
 object SparkRangerAdminPlugin extends RangerBasePlugin("spark", "sparkSql") {
 
+  /**
+   * For a Spark SQL query, it may contain 0 or more privilege objects to verify, e.g. a typical
+   * JOIN operator may have two tables and their columns to verify.
+   *
+   * This configuration controls whether to verify the privilege objects in single call or
+   * to verify them one by one.
+   */
+  def authorizeInSingleCall: Boolean =
+    getConfig.getBoolean(s"ranger.plugin.${getServiceType}.authorize.in.single.call", false)
```

Comment:
this fails ranger 0.6

Reply:
It should be alright now with added `RangerConfigProvider`. Successful built with Ranger 0.6 locally, with failed test in `show functions` case which is not changed in this PR. Tests and builds are good with higher ranger versions.